alica_engine/include/engine/IAlicaTrace.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <optional>
+#include <memory>
+
+namespace alica
+{
+
+class IAlicaTrace
+{
+public:
+    virtual ~IAlicaTrace() = default;
+    IAlicaTrace() = default;
+    IAlicaTrace(const std::string& opName, std::optional<const std::string> parent = std::nullopt);
+    IAlicaTrace(IAlicaTrace&& other) = default;
+    IAlicaTrace& operator=(IAlicaTrace&& other) = default;
+
+    virtual void setTag(const std::string& key, const std::string& value) = 0;
+    virtual void setLog(std::pair<std::string, std::string> fields) = 0;
+    virtual void markError(const std::string& description) = 0;
+    virtual std::string context() const = 0;
+
+private:
+    std::unique_ptr<IAlicaTrace> _rawTrace;
```

Comment:
an interface class should not have any member variables. The implementation will do that. Also this is a recursive reference to another IAlicaTrace which I don't quite understand.